inference-engine/src/mkldnn_plugin/nodes/ctc_greedy_decoder.cpp

```diff
@@ -0,0 +1,157 @@
+        if (layer->outData.size() != 1)
+            THROW_IE_EXCEPTION << errPrefix << "has invalid number of outputs edges: " << layer->outData.size();
+
+        auto inData = layer->insData[DATA_INDEX].lock();
+        auto sequenceLenData = layer->insData[SEQUENCE_LENGTH_INDEX].lock();
+        if (!inData || !sequenceLenData)
+            THROW_IE_EXCEPTION << errPrefix << "has nullable inputs.";
+        if (inData->getTensorDesc().getDims()[0] != sequenceLenData->getTensorDesc().getDims()[0] &&
+                inData->getTensorDesc().getDims()[1] != sequenceLenData->getTensorDesc().getDims()[1])
+            THROW_IE_EXCEPTION << errPrefix << "has invalid input shapes.";
+        if (inData->getTensorDesc().getPrecision() != Precision::FP32 &&
+                inData->getTensorDesc().getPrecision() != Precision::BF16)
+            THROW_IE_EXCEPTION << errPrefix << "has unsupported 'data' input precision: " << inData->getTensorDesc().getPrecision();
+        if (sequenceLenData->getTensorDesc().getPrecision() != Precision::FP32)
+            THROW_IE_EXCEPTION << errPrefix << "has unsupported 'sequence_length' input precision: " << sequenceLenData->getTensorDesc().getPrecision();
+
+        std::vector<DataConfigurator> inputConfigs{{ConfLayout::PLN, Precision::FP32}, {ConfLayout::PLN, Precision::FP32}};
+        std::vector<DataConfigurator> outputConfigs{{ConfLayout::PLN, Precision::FP32}};
+        addConfig(layer, inputConfigs, outputConfigs);
+
+        if (layer->CheckParamPresence("ctc_merge_repeated")) {
+            mergeRepeated_ = layer->GetParamAsBool("ctc_merge_repeated");
+        } else if (layer->CheckParamPresence("merge_repeated")) {
+            mergeRepeated_ = layer->GetParamAsBool("merge_repeated", true);
+        }
+    }
+
+    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs,
+                       ResponseDesc *resp) noexcept override {
+        const float* probabilities = inputs[DATA_INDEX]->cbuffer().as<const float*>() +
+            inputs[DATA_INDEX]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        const float* sequenceLengths = inputs[SEQUENCE_LENGTH_INDEX]->cbuffer().as<const float*>() +
+            inputs[SEQUENCE_LENGTH_INDEX]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        float* outputSequences = outputs[0]->buffer().as<float*>();
+
+        const size_t T = inputs[DATA_INDEX]->getTensorDesc().getDims()[0];
+        const size_t B = inputs[DATA_INDEX]->getTensorDesc().getDims()[1];
+        const int C = inputs[DATA_INDEX]->getTensorDesc().getDims()[2];
+        const size_t BC = B * C;
+        const size_t CB1 = C * (B - 1);
+
+        const int blankIndex = C - 1;
+
+        std::vector<size_t> sequenceLengthB(B, 0);
+        parallel_for(B, [&](size_t b) {
+            for (size_t t = 0; t < T; t++) {
+                if (sequenceLengths[B * t + b] == 0.f)
+                    break;
+                sequenceLengthB[b]++;
```

Comment:
It is not necessary to increment sequenceLengthB each time. Just do `sequenceLengthB[b] = t` after the cycle.

Reply:
Actual length could be less than T. We must increment just to the first 0.

Reply:
I mean lower_case "t"  which is incremented inside cycle exactly as "sequenceLengthB[b]". So why to increment two variables in the same manner?

Reply:
Fixed